### PR TITLE
[Chore] Refactor Storage Options

### PR DESCRIPTION
**Pull request by ashika112**
#### Description of changes
During Gen2 storage work we made some changes to naming convention for types `WithKey` and `WithPath`. Seems like we missed to update a bunch of internal storage options type. This PR updates the types name.

- Also removes an unused s3 exception file

> Note: All types updated are purely internal, no change to any external APIs

#### Description of how you validated changes
- unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Unit Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

#### Checklist for repo maintainers

- [ ] Verify E2E tests for existing workflows are working as expected or add E2E tests for newly added workflows
- [ ] New source file paths included in this PR have been added to CODEOWNERS, if appropriate

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ashwinkumar6** [comment]
Also would this conflict with `storage-browser/main` or `storage-browser/integrity`. Since it's just type rename I think it's an easy resolve

**Review by ashwinkumar6 — APPROVED** [review]

File: packages/storage/src/providers/s3/types/inputs.ts [hunk 1]
```diff
@@ -41,8 +41,8 @@ import {
  * Input type for S3 copy API.
  */
 export type CopyInput = StorageCopyInputWithKey<
```
[the remark this answers is not in the document]
Comment (ashika112) [reply packages/storage/src/providers/s3/types/inputs.ts:43]: I would keep this as this for now while we figure out how we want to approach others. Not strongly against this in this case

File: packages/storage/__tests__/providers/s3/apis/uploadData/byteLength.test.ts
Comment (HuiSF) [inline packages/storage/__tests__/providers/s3/apis/uploadData/byteLength.test.ts:1]: This unit test doesn't seem to be related to the PR title? Was this due the the test threshold complained due to other changes in this PR?

Comment (ashika112) [reply packages/storage/__tests__/providers/s3/apis/uploadData/byteLength.test.ts:1]: just added for better coverage. No issues on threshold. sorry I was making change and seems straightforward to add it in.
